src/Payments/VoucherDisplay.res

```diff
@@ -0,0 +1,110 @@
+    handlePostMessage([("iframeMountedCallback", true->Js.Json.boolean)])
+    let handle = (ev: Window.event) => {
+      let json = ev.data->Js.Json.parseExn
+      let dict = json->Utils.getDictFromJson
+      if dict->Js.Dict.get("fullScreenIframeMounted")->Belt.Option.isSome {
+        let metadata = dict->getJsonObjectFromDict("metadata")
+        let metaDataDict =
+          metadata->Js.Json.decodeObject->Belt.Option.getWithDefault(Js.Dict.empty())
+        setReturnUrl(_ => metaDataDict->getString("returnUrl", ""))
+        setDownloadUrl(_ => metaDataDict->getString("voucherUrl", ""))
+        setReference(_ => metaDataDict->getString("reference", ""))
+        setPaymentMethod(_ => metaDataDict->getString("paymentMethod", ""))
+        setPaymentIntent(_ => metaDataDict->getJsonObjectFromDict("payment_intent_data"))
+        setLoader(_ => false)
+      }
+    }
+    Window.addEventListener("message", handle)
+    Some(() => {Window.removeEventListener("message", handle)})
+  })
+
+  let closeModal = () => {
+    postSubmitResponse(~jsonData=paymentIntent, ~url=returnUrl)
+    Modal.close(setOpenModal)
+  }
+
+  <Modal showClose=false openModal setOpenModal loader>
+    <div className="flex flex-col h-full justify-between items-center">
+      <div className=" flex flex-col max-w-md justify-between items-center">
+        <div className="flex flex-row w-full">
+          <p className="Disclaimer font-medium text-sm text-[#151A1F] opacity-70">
+            {React.string(
+              `${paymentMethod->snakeToTitleCase} voucher was successfully generated! If the document hasn't started downloading automatically, click `,
+            )}
+            <a
+              className="text font-medium text-sm text-[#006DF9] underline"
+              href=downloadUrl
+              ref={linkRef->ReactDOM.Ref.domRef}
+              onClick={_ => {
+                setDownloadCounter(c => c + 1)
+                LoggerUtils.handleLogging(
+                  ~optLogger=Some(logger),
+                  ~value=downloadCounter->Js.Int.toString,
+                  ~eventName=DISPLAY_VOUCHER,
+                  ~paymentMethod,
+                  (),
+                )
+              }}>
+              {React.string("here")}
+            </a>
+            {React.string(" to download it.")}
+          </p>
+        </div>
+        <div className="flex flex-row mt-4 w-full">
+          <p className="Disclaimer font-medium text-sm text-[#151A1F] opacity-70">
+            {React.string("Bar Code Reference: ")}
+            <span className="Disclaimer font-bold text-sm text-[#151A1F] opacity-90">
+              {React.string(reference)}
+            </span>
+          </p>
+        </div>
+        <div className="Disclaimer w-full mt-16 font-medium text-xs text-[#151A1F] opacity-50">
+          {React.string(
+            " Please do not close until you have successfully downloaded the voucher, after which you will be automatically redirected. ",
```

Comment:
Please trim these spaces since React.string() will trim that automatically. Incase you want additional spaces on either side, use classNames